Codecademy/gamut
# feat(List): Ordered lists

### Overview


Makes `List` able to be rendered as an `ol` element and adds numbering to `ol` Lists.  Also adds some accessibility fixes I came across while testing. 


### PR Checklist


#### Testing Instructions



1. Head to the List storybook
2. See the new ordered list section + how nice it looks
3. See the ordered list example added to the responsive section + how nice it looks 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38127)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/7128)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
